src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -1400,10 +1400,11 @@ public interface Converter<T> {
     T convert(ResultSet rs) throws SQLException;
   }
 
-  // WARNING: SQL string must be sanitized before calling this method
-  public <T> List<T> runQuery(String sql, Converter<? extends T> converter) {
+  public <T> List<T> runQuery(
+      String query, Map<String, String> userEnteredData, Converter<? extends T> converter) {
```

Comment:
I would copy the name and type for this new parameter from the JDBC APIs, which is `NamedParameterJdbcOperations.query()`. In that case the parameter is declared as `Map<String, ?> paramMap`.
```suggestion
      String query, Map<String, ?> paramMap, Converter<? extends T> converter) {
```

It's true that right now we're using it for user entered data, but it could be used for any data that needs to be substituted.